webSockets/sender/sender.go

```diff
@@ -9,62 +9,57 @@ import (
 	"github.com/multiversx/mx-chain-core-go/core"
 	"github.com/multiversx/mx-chain-core-go/core/check"
 	"github.com/multiversx/mx-chain-core-go/core/closing"
+	"github.com/multiversx/mx-chain-core-go/webSockets"
 	"github.com/multiversx/mx-chain-core-go/webSockets/connection"
 	outportData "github.com/multiversx/mx-chain-core-go/webSockets/data"
 )
 
-var (
-	prefixWithoutAck = []byte{0}
-	prefixWithAck    = []byte{1}
-)
-
+// ArgsSender holds the arguments needed for creating a web-sockets sender
 type ArgsSender struct {
 	WithAcknowledge          bool
 	RetryDurationInSeconds   int
-	Uint64ByteSliceConverter connection.Uint64ByteSliceConverter
+	Uint64ByteSliceConverter webSockets.Uint64ByteSliceConverter
 	Log                      core.Logger
 }
 
 type sender struct {
-	counter                  uint64
-	withAcknowledge          bool
-	connections              ConnectionsHandler
-	uint64ByteSliceConverter connection.Uint64ByteSliceConverter
-	log                      core.Logger
-	safeCloser               core.SafeCloser
-	retryDuration            time.Duration
+	counter         uint64
+	withAcknowledge bool
+	connections     ConnectionsHandler
+	payloadParser   webSockets.PayloadParser
+	log             core.Logger
+	safeCloser      core.SafeCloser
+	retryDuration   time.Duration
 }
 
 func NewSender(args ArgsSender) (*sender, error) {
-	if err := checkArgs(args); err != nil {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+	payloadConverter, err := webSockets.NewWebSocketPayloadParser(args.Uint64ByteSliceConverter)
```

Comment:
for a future PR, in order to improve the code reusability, this should be injected as an argument

Reply:
done in this PR.